api/param_middleware.go

```diff
@@ -0,0 +1,66 @@
+// offers api
+// https://github.com/topfreegames/offers
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2017 Top Free Games <backend@tfgco.com>
+
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/topfreegames/offers/errors"
+	"net/http"
+)
+
+//ParamMiddleware add into the model the parameters that came in the URI
+type ParamMiddleware struct {
+	App       *App
+	Validator func(string) bool
+	Next      http.Handler
+}
+
+const paramKey = contextKey("param")
+
+//NewParamKeyMiddleware constructs a new param key middleware
+func NewParamKeyMiddleware(app *App, f func(string) bool) *ParamMiddleware {
+	m := &ParamMiddleware{App: app, Validator: f}
+	return m
+}
+
+func newContextWithParamKey(ctx context.Context, id string) context.Context {
+	return context.WithValue(ctx, paramKey, id)
+}
+
+func paramKeyFromContext(ctx context.Context) string {
+	param := ctx.Value(paramKey)
+	if param == nil {
+		return ""
+	}
+	return param.(string)
+}
+
+func (m *ParamMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
Okay, not very generic but we'll go with this for now since it won't be very hard to change in the future